### Loading the Dataset

In [ ]:
#loading the dataset
import pandas as pd
df = pd.read_csv('2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')
df.head()

### Calculating Metrics

In [33]:
print("Naively Calculated AOV: ",df['order_amount'].mean())

Naively Calculated AOV:  3145.128


In [36]:
print("Description of Total Number of Items:\n\n",df['total_items'].describe())

Description of Total Number of Items:

 count    5000.00000
mean        8.78720
std       116.32032
min         1.00000
25%         1.00000
50%         2.00000
75%         3.00000
max      2000.00000
Name: total_items, dtype: float64


There are transactions where 2000 items are being purchased. This looks like an anomaly, and hence must be looked into.

In [48]:
print("Description of Total Number of Items without transactions of 2000 items:\n\n",df[df['total_items']<2000]['total_items'].describe())

Description of Total Number of Items without transactions of 2000 items:

 count    4983.00000
mean        1.99398
std         0.98318
min         1.00000
25%         1.00000
50%         2.00000
75%         3.00000
max         8.00000
Name: total_items, dtype: float64


The maximum number of items across transactions is 8, which seems like a reasonable number. 

### Deep diving into the transactions with 2000 items purchased.

In [45]:
print("Number of Transactions with 2000 items:",len(df[(df['total_items'] == 2000)]))
print("Shop id from which 2000 items were purchased:",df[(df['total_items'] == 2000)]['shop_id'].unique()[0].astype(int))

Number of Transactions with 2000 items: 17
Shop id from which 2000 items were purchased: 42


###  Top 5 Shops with highest revenue

In [50]:
df.groupby('shop_id').agg({'order_amount':'sum'}).sort_values('order_amount',ascending = False).iloc[0:5,]

order_amount
shop_id              
42           11990176
78            2263800
89              23128
81              22656
6               22627

The high revenue from shop 42 is attributed to the high quantity transactions from the shop. Interestingly, shop 78 also has unusually high revenue. 

### Deep diving into shop 78

In [54]:
#Calculating average price per sneaker
df['price_per_sneaker'] = df['order_amount']/df['total_items']

print("Average price per sneaker in shop 78:",df[df['shop_id'] == 78]['price_per_sneaker'].mean())

Average price per sneaker in shop 78: 25725.0


$25,725 is an unusually high amount for the average price of a sneaker. 

In [57]:
df.groupby('shop_id').agg({'price_per_sneaker':'mean'}).sort_values('price_per_sneaker',ascending = False).iloc[0:5,]

price_per_sneaker
shop_id                   
78                 25725.0
42                   352.0
12                   201.0
89                   196.0
99                   195.0

### Calculating New AOV excluding transactions with 2000 items and shop 78

In [58]:
df[(df['total_items'] < 2000) & (df['shop_id'] != 78)]['order_amount'].mean()

302.58051448247926

### Calculating the most sold item

In [30]:
df.groupby('shop_id').agg({'total_items':'sum'}).sort_values('total_items',ascending = False)

total_items
shop_id             
42             34063
13               136
84               132
71               130
53               130
...              ...
38                72
16                71
44                71
56                69
85                67

[100 rows x 1 columns]

### Calculating the least sold item

In [59]:
df.groupby('shop_id').agg({'total_items':'sum'}).sort_values('total_items')

total_items
shop_id             
85                67
56                69
16                71
44                71
38                72
...              ...
53               130
71               130
84               132
13               136
42             34063

[100 rows x 1 columns]